### ISSUE TITLE
Implement better error handling

### ISSUE THREAD

**Issue by gahujipo** [Isaksson/node-red-contrib-unifi#46]
I am not able to get this node running. I run my unifi controller under a URL and an invalid SSL certificate (comes by default). But no matter which command I choose I get this error: 
```
"TypeError: Cannot read property 'command' of undefined"
```

This is my node-red: 
![image](https://user-images.githubusercontent.com/26710574/93642029-ab0cb400-f9fd-11ea-9508-c2eadbb93072.png)

These are the versions in my controller: 
![image](https://user-images.githubusercontent.com/26710574/93642631-abf21580-f9fe-11ea-8cc2-760c6654e5eb.png)

What could I check next?

**Comment by Isaksson** [comment]
What do you inject from the inject node?

**Comment by gahujipo** [comment]
Nothing. Inject is empty. Do I have to inject the command? Shouldn't that be selected in the drop-down of the node?

**Comment by gahujipo** [comment]
I just upgraded to the latest v5 controller version. Didn't change anything. Is this node already compatible with controllers with version 6?

**Comment by gahujipo** [comment]
Is there a log I could check for errors to get this node running?

**Comment by Isaksson** [comment]
Does it work if you actually injects a command?

**Comment by gahujipo** [comment]
No. I changed the inject node to this:
![grafik](https://user-images.githubusercontent.com/26710574/94544322-637ef700-024b-11eb-8651-1537998dbe0a.png)

Which leads to this with **no** debug output: 
![grafik](https://user-images.githubusercontent.com/26710574/94544574-b193fa80-024b-11eb-83cb-795e45f29873.png)

**Comment by Isaksson** [comment]
I need to add some better error handling but one of the issues could be wrong IP or Port number, then it will give you that object Object error.

**Comment by gahujipo** [comment]
Yes, I think it would be the best solution to add some better error handling. 

Like I said, I am running the controller under a URL (controller.domain.tld, not an IP) with the default self signed ceritificate. Could that be the problem which prevents the node to connect to the controller?

**Comment by gahujipo** [comment]
I found in the source code, that the error, gets written to the console, which I was able to get from the container console. There I found 
```
ERROR: Error: getaddrinfo ENOTFOUND https https:443
```
I changed the server property to a value without a leading "https://", which made the node green, but got again the error 
```
"TypeError: Cannot read property 'command' of undefined"
```
![grafik](https://user-images.githubusercontent.com/26710574/94577740-2fb9c680-0277-11eb-8886-398e4f0b6960.png)

Then I injected the command, as suggested: 
```JSON
{
    "command": "ClientDevices"
}
```

And now it works as expected. Therefore I will change the title of this issue.

**Comment by hannah98** [comment]
I don't know if it is appropriate to latch on to this issue but I discovered a similar issue that seems to fit under the "better error handling" message.

I tried using this node for the first time today.  I set it up and whenever I would run it the only indication I would get that it was doing something was this message in the system console:
```
ERROR: undefined
```

I tried injecting the command as shown above and that didn't work.  Finally through trial and error I discovered what the problem was.

The name of my Unifi site is "Home", so I changed the Site in this node configuration from "default" to "Home".  That was producing the ```ERROR: undefined```.

As soon as I changed the site back to "default" it started working.  It would have been helpful to know that the problem had to do with the value I entered into the site field.
